Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -66,60 +76,68 @@ public final class ViewAnnotationManager {
+    public func remove(_ view: UIView) throws {
+        guard let id = idsByView[view], let annotatonView = viewsById[id] else {
             return
         }
         try mapboxMap.removeViewAnnotation(withId: id)
         annotatonView.removeFromSuperview()
-        annotationViewsById.removeValue(forKey: id)
-        annotationIdByView.removeValue(forKey: annotatonView)
+        viewsById.removeValue(forKey: id)
+        idsByView.removeValue(forKey: annotatonView)
     }
 
-    /// Update given `AnnotationView` with `ViewAnnotationOptions`.
+    /// Update given `UIView` with `ViewAnnotationOptions`.
     /// Important thing to keep in mind that only properties present in `options` will be updated,
     /// all other will remain the same as specified before.
     ///
     /// - Parameters:
-    ///   - annotatonView: `AnnotationView` to be updated
+    ///   - view: `UIView` to be updated
     ///   - options: view annotation options with optional fields used for the update
     ///
     /// - Throws:
     ///   - `ViewAnnotationManagerError.annotationNotFound`: the supplied view was not found
     ///   - `MapError`: errors during the update of the view (eg. incorrect parameters)
-    public func update(_ annotatonView: AnnotationView, options: ViewAnnotationOptions) throws {
-        guard let id = annotationIdByView[annotatonView] else {
+    public func update(_ view: UIView, options: ViewAnnotationOptions) throws {
+        guard let id = idsByView[view] else {
             throw ViewAnnotationManagerError.annotationNotFound
         }
         try mapboxMap.updateViewAnnotation(withId: id, options: options)
+        let isHidden = !(options.visible ?? true)
+        expectedHiddenForId[id] = isHidden
+        viewsById[id]?.isHidden = isHidden
     }
 
-    /// Find `AnnotationView` by feature id if it was specified as part of `ViewAnnotationOptions.associatedFeatureId`.
+    /// Find `UIView` by feature id if it was specified as part of `ViewAnnotationOptions.associatedFeatureId`.
     ///
     /// - Parameters:
-    ///   - identifier: the identifier of the feature which will be used for finding the associated `AnnotationView`
+    ///   - identifier: the identifier of the feature which will be used for finding the associated `UIView`
     ///
-    /// - Returns: `AnnotationView` if view was found and `nil` otherwise.
-    public func viewAnnotation(forFeatureId identifier: String) -> AnnotationView? {
-        return annotationViewsById.keys.first(where: { id in
+    /// - Returns: `UIView` if view was found and `nil` otherwise.
+    public func view(forFeatureId identifier: String) -> UIView? {
+        return viewsById.keys.first(where: { id in
             (try? mapboxMap.options(forViewAnnotationWithId: id).associatedFeatureId == identifier) ?? false
         }).flatMap({
-            annotationViewsById[$0]
+            viewsById[$0]
         })
     }
```

Comment:
We should profile this. Seems like it could be pretty slow.

Reply:
True, and I was thinking to cache the `associatedFeatureId` of the annotations. (Maybe as another key-value pair.)
Internally there's another loop, so if we have a lot of annotations it could get slow: https://github.com/mapbox/mapbox-gl-native-internal/blob/e0253c010a42b7e57dc17331cb47234e41a428aa/internal/src/mapbox/maps/view_annotation_manager.cpp#L307

However it was a decided in the Android implementation to use the underlying functions to query the options, because that way there's a single source of truth for the options.
Personally I think we could make an exception for this to speed up these functions.

Reply:
Or maybe add a method to GL native that allows getting just the feature id to reduce the amount of serialization/deserialization that has to happen

Reply:
I also like your idea of caching the feature IDs. Reducing round trips through GL Native is probably going to be the fastest.

Reply:
I think I'll go with the caching, just because it keeps the underlying API more simple. But we can change it later if needed.

Reply:
And would be great to optimize the GL Native impl too to avoid the looping there. If the expected usage pattern is key-based, wouldn't it make sense to use a key-based lookup structure instead of a list?

Reply:
Yes we had a discussion about that. If I remember correctly the problem was that `std::map` is unordered, but we need to keep them ordered for the collision detection. Miao said she will explore some other options to optimize it later on.